core/src/banking_stage.rs

```diff
@@ -1146,8 +1214,19 @@ impl BankingStage {
         );
         inc_new_counter_debug!("banking_stage-transactions_received", count);
         let mut proc_start = Measure::start("process_packets_transactions_process");
+        let mut vote_tx_count = 0;
         let mut new_tx_count = 0;
 
+        // process all votes in the pipeline first
+        Self::process_packets_vote(
+            &mut mms,
+            &mut vote_tx_count,
+            poh,
+            recorder,
+            transaction_status_sender.clone(),
+            gossip_vote_sender,
+        );
+
```

Comment:
@sakridge maybe can add `vote-only` feature at here. This is the point where votes in pipeline have been processed, and about to process/forward other txs in received packets. It could check `bank.vote_only_bank()` and skip next step if necessary. wdyt?

Reply:
yea, that seems ok.

Reply:
That is reasonable. I'm starting to be a bit worried about the code duplication we introduce in this way. We'll have to have this process_votes on the buffered packet processing as well, right? And we would deserialize everything twice except votes.

Also, do we need some of the other checks that we make in packets_from_transactions?

Reply:
Yeah, the duplication is a concern, folding vote-only logic here avoids some. I'm copying your code from #19916 to here. 

Not sure if necessary to do `process_packets_vote` again on `process_buffered_packets` again. Once votes are processed at point of receiving, they will not being put into buffer, except when `! get_bank_still_processing_txs`. Not sure how often this happens, will add a metric to check it out.

Reply:
Yes, they can be received prior to the validator being the leader.

Reply:
So then it will process the buffered ones first. I think we need to handle both those pathways.

Reply:
Gotcha, that explains the metric. Receiving packets before becoming leader is solid need for apply same QoS on buffered. Thanks for pointing it out!